indico/modules/events/registration/controllers/display.py

```diff
@@ -92,6 +92,8 @@ def _process_args(self):
                 raise NotFound
         else:
             self.registration = self.regform.get_registration(user=session.user) if session.user else None
+            if self.regform.is_hidden and self.regform.uuid != request.args.get('form_token'):
+                raise NotFound
```

Comment:
Do not check this if we have a `self.registration`

Reply:
Done.